{
    "containers": {
        "cna": {
            "affected": [
                {
                    "product": "vm2",
                    "vendor": "n/a",
                    "versions": [
                        {
                            "lessThan": "3.6.11",
                            "status": "affected",
                            "version": "unspecified",
                            "versionType": "custom"
                        }
                    ]
                }
            ],
            "credits": [
                {
                    "lang": "en",
                    "value": "XmiliaH"
                }
            ],
            "datePublic": "2022-07-13T00:00:00",
            "descriptions": [
                {
                    "lang": "en",
                    "value": "This affects the package vm2 before 3.6.11. It is possible to trigger a RangeError exception from the host rather than the \"sandboxed\" context by reaching the stack call limit with an infinite recursion. The returned object is then used to reference the mainModule property of the host code running the script allowing it to spawn a child_process and execute arbitrary code."
                }
            ],
            "metrics": [
                {
                    "cvssV3_1": {
                        "attackComplexity": "LOW",
                        "attackVector": "NETWORK",
                        "availabilityImpact": "LOW",
                        "baseScore": 8.3,
                        "baseSeverity": "HIGH",
                        "confidentialityImpact": "LOW",
                        "exploitCodeMaturity": "PROOF_OF_CONCEPT",
                        "integrityImpact": "LOW",
                        "privilegesRequired": "NONE",
                        "remediationLevel": "OFFICIAL_FIX",
                        "reportConfidence": "CONFIRMED",
                        "scope": "CHANGED",
                        "temporalScore": 7.5,
                        "temporalSeverity": "HIGH",
                        "userInteraction": "NONE",
                        "vectorString": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:C/C:L/I:L/A:L/E:P/RL:O/RC:C",
                        "version": "3.1"
                    }
                }
            ],
            "problemTypes": [
                {
                    "descriptions": [
                        {
                            "description": "Sandbox Bypass",
                            "lang": "en",
                            "type": "text"
                        }
                    ]
                }
            ],
            "providerMetadata": {
                "dateUpdated": "2022-07-13T08:20:14",
                "orgId": "bae035ff-b466-4ff4-94d0-fc9efd9e1730",
                "shortName": "snyk"
            },
            "references": [
                {
                    "tags": [
                        "x_refsource_MISC"
                    ],
                    "url": "https://snyk.io/vuln/SNYK-JS-VM2-473188"
                },
                {
                    "tags": [
                        "x_refsource_MISC"
                    ],
                    "url": "https://github.com/patriksimek/vm2/issues/197"
                },
                {
                    "tags": [
                        "x_refsource_MISC"
                    ],
                    "url": "https://github.com/patriksimek/vm2/commit/4b22d704e4794af63a5a2d633385fd20948f6f90"
                }
            ],
            "title": "Sandbox Bypass"
        }
    },
    "cveMetadata": {
        "assignerOrgId": "bae035ff-b466-4ff4-94d0-fc9efd9e1730",
        "assignerShortName": "snyk",
        "cveId": "CVE-2019-10761",
        "dateReserved": "2019-04-03T00:00:00",
        "dateUpdated": "2022-07-13T08:20:14",
        "state": "PUBLISHED",
        "datePublished": "2022-07-13T08:20:14.533113Z"
    },
    "dataType": "CVE_RECORD",
    "dataVersion": "5.0"
}
